doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,500 @@
+# Implementing a face beautification algorithm on G-API {#tutorial_gapi_face_beautification}
+
+@prev_tutorial{tutorial_gapi_anisotropic_segmentation}
+
+[TOC]
+
+# Introduction {#gapi_fb_intro}
+
+In this tutorial you will learn:
+* How to implement a custom kernel (using OpenCV backend);
+* How to use Inference-API to infer networks inside the graph;
+* How to process a videostream inside the graph using Streaming-API;
+* The main ideas of the Face Beautification algorithm.
+
+You can find source code in the `modules/gapi/samples/face_beautification.cpp`
+of the OpenCV source code library.
+
+# New G-API features description {#gapi_fb_features}
+
+Before we can discuss an implementation of any algorithm, we should
+sort out new G-API features applied there. Obviously, the paragraphs below are
+not detailed documentary articles, they are just brief guides how to use one
+feature or another with real, living examples given.
+
+## Custom kernels implementation {#gapi_fb_custom_kernels}
+
+A new kernel implementation has been already described clearly in OpenCV G-API
+documentation (see [G-API Kernel API](@ref gapi_kernel_api)). Bare steps will
+be mentioned here with few details. In simple words, to implement a brand
+new G-API kernel we have to take 3 main steps:
+
+### 1) New kernel interface definition {#gapi_fb_c_k_1}
+
+There is a couple of macros to define a new kernel. The first one is
+`G_TYPED_KERNEL()`, which declares a new operation properties:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_decl
+
+To get the full description, see [Defining a kernel](@ref gapi_defining_kernel).
+
+A small yet significant adjustment has to be made when a kernel with a multiple
+return value is defined. For this case, we must use another macro,
+`G_TYPED_KERNEL_M()`:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_m_decl
+
+As you can see, almost everything is unchanged (speaking of the initial
+look, of course). Note: the `std::tuple` appearance in a kernel signature could
+be the sign to change the macro.
+
+The `outMeta()` function for such kernels must return a tuple too.
+
+### 2) Kernel implementationfor a specific backend. {#gapi_fb_c_k_2}
+
+Similarly, the second step begins with a macro. If we want to implement a CPU
+kernel version using OpenCV functions, we must use the `GAPI_OCV_KERNEL()`
+macro, which defines what the operation should do:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_impl
+
+For more details (including arguments of the function `run()` and their order),
+visit [Implementing a kernel](@ref gapi_kernel_implementing).
+
+Note: all the output arguments of a kernel signature
+are just put after input ones keeping the order if they are multiple:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification_stream.cpp kern_m_impl
```

Comment:
I'd keep it off this tutorial (or may be put as a last chapter with no big emphasis on it). The key of this tutorial is to show how to build a hybrid CV/DL compute graph, and custom kernels with `outMeta` is just a tiny specific detail of it.